tests/phpunit/TemplateTest.php

```diff
@@ -1657,6 +1657,15 @@ public function testTrimResearchGateETC() {
     $this->assertEquals('https://www.google.com/search?hl=en&safe=off&q=%22west+coast+hotel+co.+v.+parrish%22+(site%3Anewsweek.com+OR+site%3Apost-gazette.com+OR+site%3Ausatoday.com+OR+site%3Awashingtonpost.com+OR+site%3Atime.com+OR+site%3Areuters.com+OR+site%3Aeconomist.com+OR+site%3Amiamiherald.com+OR+site%3Alatimes.com+OR+site%3Asfgate.com+OR+site%3Achicagotribune.com+OR+site%3Anytimes.com+OR+site%3Awsj.com+OR+site%3Ausnews.com+OR+site%3Amsnbc.com+OR+site%3Anj.com+OR+site%3Atheatlantic.com)', $prepared->get('url'));
   }
  
+  public function testCleanRGTitles() {
+    $text = '{{cite journal|url=http://researchgate.net/publication/320041870_yup|title=Hello {{!}} Request PDF}}';
```

Comment:
I suppose I'm wondering how this {{!}} gets into the title – is the bot converting a `|` when it finds one, or is this the work of some other bot?  If the former, perhaps a more useful test would be one where we give the bot a URL from which it extracts the title "Hello | Request PDF", and ends up with "Hello"?